rfc/text/advanced/ap_27_pubsub_event_history.md

```diff
@@ -1,58 +1,132 @@
 ## Event History {#pubsub-event-history}
 
-Instead of complex QoS for message delivery, a *Broker* may provide *message history*. A *Subscriber* is responsible to handle overlaps (duplicates) when it wants "exactly-once" message processing across restarts.
-
-The *Broker* may allow for configuration on a per-topic basis.
+Instead of complex QoS for message delivery, a *Broker* may provide *message history*. A *Subscriber* is 
+responsible to handle overlaps (duplicates) when it wants "exactly-once" message processing across restarts.
 
 The event history may be transient or persistent message history (surviving *Broker* restarts).
 
-A *Broker* that implements *event history* must (also) announce role `HELLO.roles.callee`, indicate `HELLO.roles.broker.history == 1` and provide the following (builtin) procedures.
+The *Broker* implementation may allow for configuration on a per-topic basis.
+
+To understand what actually event history means lets review the publication flow. When one peer decides to publish
+a message to a topic it results in `Publication` WAMP message with concrete `publication_id`, `payload` and `options`
+among other attributes. The *Broker* in his turn checks current subscriptions and create an `Event` WAMP message for
+every subscription. There can be no subscriptions, one exactly matched to publication topic subscription and zero or 
```

Comment:
"""
A given event received by the router from a publisher (a `PUBLISH` message) will lead to one or more matching subscriptions:
* zero or one exact subscription
* zero or more prefix subscriptions
* zero or more wildcard subscriptions

One and the same published event is dispatched and sent to every subscriber of all subscriptions matching. Thus, a given event might be sent multiple times to the same client under different subscriptions.
"""

Reply:
Added!